pkg/controllers/scheduling/scheduling_controller.go

```diff
@@ -238,16 +238,8 @@ func (c *schedulingController) syncPlacement(ctx context.Context, syncCtx factor
 		status,
 	)
 
-	// update status and return error if schedule() returns error
-	if status.IsError() {
-		if err := c.updateStatus(ctx, placement, int32(len(scheduleResult.Decisions())), misconfiguredCondition, satisfiedCondition); err != nil {
-			return err
-		}
-		return status.AsError()
-	}
-
 	// requeue placement if requeueAfter is defined in scheduleResult
-	if syncCtx != nil && scheduleResult.RequeueAfter() != nil {
+	if syncCtx != nil && !status.IsError() && scheduleResult.RequeueAfter() != nil {
```

Comment:
I think we should requeue no matter there is error or not? and should we return error finally after update status?